firewalld/zones.sls

```diff
@@ -44,3 +44,16 @@ directory_firewalld_zones:
         zone: {{ v|json }}
 
 {% endfor %}
+
+{%- if firewalld.get('purge_zones', False) %}
+{%- for file in salt['file.find']('/etc/firewalld/zones', name='*.xml', print='name', type='f') %}
+
+{%- if file.replace('.xml', '') not in firewalld.get('zones', {}).keys() %}
```

Comment:
As a matter of style, I would set a variable to calculate the `firewalld.get('zones', {}).keys()` once instead of for each file.

Reply:
Hi, I concur. I switched the existing loop in the file to a variable as well whilst at it, see 7883127.